router.go

```diff
@@ -52,16 +55,20 @@ type MapRouter struct {
 	serviceProcedures         map[serviceProcedure]transport.Procedure
 	serviceProcedureEncodings map[serviceProcedureEncoding]transport.Procedure
 	supportedEncodings        map[serviceProcedure][]string
+	serviceNames              map[string]struct{}
 }
 
 // NewMapRouter builds a new MapRouter that uses the given name as the
 // default service name.
-func NewMapRouter(defaultService string) MapRouter {
+func NewMapRouter(defaultService string) (m MapRouter) {
```

Comment:
nit: why name the return value here and not use it - seems like an unnecessary delta

Reply:
sorry, forgot to remove it after several iterations of review.